plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -264,37 +365,53 @@ class Bazel(
             }
         }.getOrNull()
 
-    private fun getDependencyGraph(projectDir: File, depDirectives: Map<String, BazelDepDirective>): Set<Scope> {
+    private fun getDependencyGraph(
+        projectDir: File,
+        depDirectives: Map<String, BazelDepDirective>,
+        archiveOverrides: Map<String, ArchiveOverride>
+    ): Set<Scope> {
         val process = run("mod", "graph", "--output", "json", "--disk_cache=", workingDir = projectDir)
         val mainModule = process.stdout.parseBazelModule()
         val (mainDeps, devDeps) = mainModule.dependencies.partition { depDirectives[it.key]?.devDependency != true }
 
         return setOf(
             Scope(
                 name = "main",
-                dependencies = mainDeps.mapTo(mutableSetOf()) { it.toPackageReference() }
+                dependencies = mainDeps.mapTo(mutableSetOf()) { it.toPackageReference(archiveOverrides) }
             ),
             Scope(
                 name = "dev",
-                dependencies = devDeps.mapTo(mutableSetOf()) { it.toPackageReference() }
+                dependencies = devDeps.mapTo(mutableSetOf()) { it.toPackageReference(archiveOverrides) }
             )
         )
     }
 
-    private fun BazelModule.toPackageReference(): PackageReference =
-        PackageReference(
+    /**
+     * Convert a [BazelModule] to a [PackageReference]. If an archive override is present in [archiveOverrides], the
+     * version of the package will be removed as Bazel does in the output of "mod graph".
+     */
+    private fun BazelModule.toPackageReference(archiveOverrides: Map<String, ArchiveOverride>): PackageReference {
+        val name2 = name ?: key.substringBefore("@", "")
+        val version2 = if (archiveOverrides.containsKey(name2)) {
```

Comment:
Prefer Kotlin's `in` operator.

Reply:
Maybe something like:

```
val version2 = = archiveOverrides.takeUnless { name2 in it }?.let { version ?: key.substringAfter("@", "") }.orEmpty()
```

Reply:
Can you thing of better variable names than just appending a "2"? Maybe append an "override" prefix instead?